src/test/java/com/amazonaws/encryptionsdk/internal/BlockEncryptionHandlerTest.java

```diff
@@ -67,4 +83,88 @@ public void correctIVGenerated() throws Exception {
         },
         headers.getNonce());
   }
+
+  /**
+   * This isn't a unit test, but it reproduces a bug found in the FrameEncryptionHandler where the
+   * stream would be truncated when the offset is >0. For the sake of robustness, the same test is
+   * included against the BlockEncryptionHandler.
+   *
+   * @throws Exception
+   */
+  @Test
+  public void testStreamTruncation() throws Exception {
+    // Initialize AES key and keyring
+    SecureRandom rnd = new SecureRandom();
+    byte[] rawKey = new byte[16];
+    rnd.nextBytes(rawKey);
+    SecretKeySpec cryptoKey = new SecretKeySpec(rawKey, "AES");
+    MaterialProviders materialProviders =
+        MaterialProviders.builder()
+            .MaterialProvidersConfig(MaterialProvidersConfig.builder().build())
+            .build();
+    CreateRawAesKeyringInput keyringInput =
+        CreateRawAesKeyringInput.builder()
+            .wrappingKey(ByteBuffer.wrap(cryptoKey.getEncoded()))
+            .keyNamespace("Example")
+            .keyName("RandomKey")
+            .wrappingAlg(AesWrappingAlg.ALG_AES128_GCM_IV12_TAG16)
+            .build();
+    IKeyring keyring = materialProviders.CreateRawAesKeyring(keyringInput);
+    // Use unframed (block) encryption
+    AwsCrypto crypto = AwsCrypto.builder().withEncryptionFrameSize(0).build();
+
+    String testDataString = StringUtils.repeat("Hello, World! ", 5_000);
+
+    int startOffset = 100; // The data will start from this offset
+    byte[] inputDataWithOffset = new byte[10_000];
+    // the length of the actual data
+    int dataLength = inputDataWithOffset.length - startOffset;
+    // copy some data, starting at the startOffset
+    // so the first |startOffset| bytes are 0s
+    System.arraycopy(
+        testDataString.getBytes(StandardCharsets.UTF_8),
+        0,
+        inputDataWithOffset,
+        startOffset,
+        dataLength);
+    // decryptData (non-streaming) doesn't know about the offset
+    // it will strip out the original 0s
+    byte[] expectedOutput = new byte[10_000 - startOffset];
```

Comment:
nit for clarity, maybe

```suggestion
    byte[] expectedOutputForNonStreamDecrypt = new byte[10_000 - startOffset];
```